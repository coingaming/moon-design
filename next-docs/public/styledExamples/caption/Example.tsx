import { Caption } from '@heathmont/moon-core';

const Example = () => <Caption>Caption</Caption>;

export default Example;
