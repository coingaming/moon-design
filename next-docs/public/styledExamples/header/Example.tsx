import React from 'react';
import { Header } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex justify-around items-center w-full">
    <Header>Title</Header>
  </div>
);

export default Example;
