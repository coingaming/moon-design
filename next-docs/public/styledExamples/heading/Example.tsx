import { Heading } from '@heathmont/moon-core';

const Example = () => <Heading>Heading</Heading>;

export default Example;
