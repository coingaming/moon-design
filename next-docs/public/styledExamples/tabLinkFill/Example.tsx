import { TabLinkFill } from '@heathmont/moon-core';

const Example = () => <TabLinkFill />;

export default Example;
