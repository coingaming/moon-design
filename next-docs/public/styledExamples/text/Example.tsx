import { Text } from '@heathmont/moon-core';

const Example = () => <Text>Text</Text>;

export default Example;
