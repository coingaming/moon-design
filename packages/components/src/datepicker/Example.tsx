import * as React from 'react';
import { enGB } from 'date-fns/locale';

import DateRangePicker from './DateRangePicker';

const Example = () => {
  const [initialStartDate, setStartDate] = React.useState(
    new Date('September 17, 2020 03:24:00')
  );
  const [initialEndDate, setEndDate] = React.useState(
    new Date('September 27, 2020 03:24:00')
  );

  return (
    <>
      <DateRangePicker
        startDate={initialStartDate}
        endDate={initialEndDate}
        onDateChange={({ startDate, endDate }: any) => {
          setStartDate(startDate);
          setEndDate(endDate);
        }}
        config={{
          weekStartsOn: 1,
          withHoursAndMinutes: true,
          locale: enGB,
        }}
      />
    </>
  );
};

export default Example;
