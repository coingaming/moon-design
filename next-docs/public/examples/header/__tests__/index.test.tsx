/**
 * @jest-environment jsdom
 */

import React from 'react';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';
import renderer from 'react-test-renderer';

import Example from '../Example';
import BackButton from '../BackButton';
import CloseButton from '../CloseButton';
import Icons from '../Icons';
import IsDivider from '../IsDivider';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

describe('Header', () => {
  it('renders', () => {
    const testRenderer = renderer.create(renderWithTheme(<Example />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with back button', () => {
    const testRenderer = renderer.create(renderWithTheme(<BackButton />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with close button', () => {
    const testRenderer = renderer.create(renderWithTheme(<CloseButton />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with icons', () => {
    const testRenderer = renderer.create(renderWithTheme(<Icons />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with is divider', () => {
    const testRenderer = renderer.create(renderWithTheme(<IsDivider />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
