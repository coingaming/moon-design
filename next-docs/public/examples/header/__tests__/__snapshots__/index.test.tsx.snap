// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  className="flex justify-around items-center w-full"
>
  <div
    className="sc-iXLiCy kcdsdY"
  >
    <div
      className="sc-ixlHCg gxPqkx"
    >
      Title
    </div>
    <div
      className="sc-kEDQaP BGkxu"
    />
  </div>
</div>
`;

exports[`Header renders with back button 1`] = `
<div
  className="flex justify-around items-center w-full"
>
  <div
    className="sc-iXLiCy kcdsdY"
  >
    <div
      className="sc-ednYOV hRGdKm"
    >
      <svg
        className="sc-ibkoSk"
        height="1em"
        viewBox="0 0 20 8"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.08 8V4.8H20V3.2H5.08V0L0 4z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="sc-ixlHCg gxPqkx"
    >
      Title
    </div>
    <div
      className="sc-kEDQaP BGkxu"
    />
  </div>
</div>
`;

exports[`Header renders with close button 1`] = `
<div
  className="flex justify-around items-center w-full"
>
  <div
    className="sc-iXLiCy kcdsdY"
  >
    <div
      className="sc-ixlHCg gxPqkx"
    >
      Title
    </div>
    <div
      className="sc-kEDQaP BGkxu"
    >
      <div
        className="sc-izfXps uhWsK"
      >
        <svg
          className="sc-gVIfzB"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.609 10L20 17.391 17.391 20 10 12.609 2.609 20 0 17.391 7.391 10 0 2.609 2.609 0 10 7.391 17.391 0 20 2.609z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header renders with icons 1`] = `
<div
  className="flex justify-around items-center w-full"
>
  <div
    className="sc-iXLiCy kcdsdY"
  >
    <div
      className="sc-ixlHCg gxPqkx"
    >
      Title
    </div>
    <div
      className="sc-kEDQaP BGkxu"
    >
      <span
        className="sc-keepax geygmX"
      >
        <svg
          className="sc-eA-DzvL"
          fill="none"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M9.981 0C4.467 0 0 4.472 0 9.994c0 .325.012.65.05.974.012.113.037.225.05.35.025.187.05.387.087.587.025.138.063.275.1.425.037.162.075.325.125.475.05.15.1.312.15.462.05.137.1.262.15.4a8.229 8.229 0 00.373.812c.088.174.175.337.275.5.05.087.1.162.15.249l.337.525c.037.05.087.112.124.162a9.95 9.95 0 004.779 3.523.597.597 0 01.087.037c.462.15.936.275 1.423.363.087.012.162.025.25.037.486.075.985.125 1.496.125.512 0 1.01-.05 1.497-.125.088-.012.163-.025.25-.037.487-.088.96-.2 1.422-.363.038-.012.063-.025.088-.037a10.029 10.029 0 004.778-3.523c.038-.05.075-.1.113-.162.124-.175.236-.35.349-.525.05-.088.1-.162.15-.25.1-.162.187-.337.274-.512.05-.1.1-.212.15-.325.075-.162.137-.325.212-.487.05-.125.1-.262.15-.4.05-.15.1-.3.15-.462.05-.162.087-.312.124-.475.037-.137.075-.274.1-.424.037-.2.062-.388.087-.588.013-.112.038-.224.05-.35.025-.324.05-.636.05-.974C19.963 4.472 15.496 0 9.981 0zm0 17.989a7.954 7.954 0 01-6.138-2.886c.836-.45 1.859-.824 2.67-1.187 1.447-.65 1.297-1.049 1.347-1.586.013-.075.013-.138.013-.213-.512-.45-.924-1.074-1.198-1.798v-.013c0-.012-.012-.025-.012-.025a3.634 3.634 0 01-.15-.487c-.35-.063-.55-.437-.624-.787a1.57 1.57 0 01-.187-.862c.05-.512.25-.737.474-.837v-.075c0-.637.062-1.549.175-2.149.025-.162.062-.324.112-.487a3.64 3.64 0 011.26-1.861c.611-.475 1.485-.737 2.27-.737.774 0 1.648.25 2.271.737a3.64 3.64 0 011.26 1.861c.05.163.088.325.113.5.112.6.175 1.511.175 2.148v.088c.224.1.411.325.461.824.038.388-.124.725-.2.862-.074.338-.261.712-.598.775a2.997 2.997 0 01-.15.475c0 .012-.012.05-.012.05a4.462 4.462 0 01-1.173 1.773c0 .075.012.163.012.238.05.537-.15.937 1.31 1.586.811.363 1.834.75 2.67 1.187a8 8 0 01-6.15 2.886z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="sc-keepax geygmX"
      >
        <svg
          className="sc-futMm"
          fill="none"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 1a9 9 0 00-9 9 9 9 0 009 9 9 9 0 009-9 9 9 0 00-9-9zm0 17.1A8.1 8.1 0 011.9 10 8.1 8.1 0 0110 1.9a8.1 8.1 0 018.1 8.1 8.1 8.1 0 01-8.1 8.1zm0-9.9a.424.424 0 01.45.45v4.95h1.35c.6 0 .6.9 0 .9H8.2c-.6 0-.6-.9 0-.9h1.35V9.1H9.1c-.6 0-.6-.9 0-.9h.9zm.9-1.8a.9.9 0 10-1.8 0 .9.9 0 001.8 0z"
            fill="currentColor"
            fillRule="evenodd"
          />
          <path
            d="M10 8.2v.1h.002L10 8.2zm.45 5.4h-.1v.1h.1v-.1zm-.9 0v.1h.1v-.1h-.1zm0-4.5h.1V9h-.1v.1zm-.002 0v.1-.1zM1.1 10A8.9 8.9 0 0110 1.1V.9A9.1 9.1 0 00.9 10h.2zm8.9 8.9A8.9 8.9 0 011.1 10H.9a9.1 9.1 0 009.1 9.1v-.2zm8.9-8.9a8.9 8.9 0 01-8.9 8.9v.2a9.1 9.1 0 009.1-9.1h-.2zM10 1.1a8.9 8.9 0 018.9 8.9h.2A9.1 9.1 0 0010 .9v.2zM1.8 10a8.2 8.2 0 008.2 8.2V18a8 8 0 01-8-8h-.2zM10 1.8A8.2 8.2 0 001.8 10H2a8 8 0 018-8v-.2zm8.2 8.2A8.2 8.2 0 0010 1.8V2a8 8 0 018 8h.2zM10 18.2a8.2 8.2 0 008.2-8.2H18a8 8 0 01-8 8v.2zm.55-9.55a.541.541 0 00-.173-.416.548.548 0 00-.38-.134l.005.2a.349.349 0 01.242.083c.061.054.106.14.106.267h.2zm0 4.95V8.65h-.2v4.95h.2zm1.25-.1h-1.35v.2h1.35v-.2zm0 1.1a.535.535 0 00.412-.174.566.566 0 00.138-.376.566.566 0 00-.138-.376.535.535 0 00-.412-.174v.2c.123 0 .208.045.263.107a.366.366 0 01.087.243c0 .091-.03.18-.087.243a.336.336 0 01-.263.107v.2zm-3.6 0h3.6v-.2H8.2v.2zm0-1.1a.535.535 0 00-.412.174.566.566 0 00-.138.376c0 .134.044.27.138.376a.535.535 0 00.412.174v-.2a.336.336 0 01-.263-.107.366.366 0 01-.087-.243c0-.091.03-.18.087-.243A.336.336 0 018.2 13.7v-.2zm1.35 0H8.2v.2h1.35v-.2zm-.1-4.4v4.5h.2V9.1h-.2zm.098.1h.002V9h-.002v.2zm-.448 0h.448V9H9.1v.2zm0-1.1a.535.535 0 00-.412.174.566.566 0 00-.138.376c0 .134.044.27.138.376A.535.535 0 009.1 9.2V9a.336.336 0 01-.263-.107.366.366 0 01-.087-.243c0-.091.03-.18.087-.243A.336.336 0 019.1 8.3v-.2zm.9 0h-.9v.2h.9v-.2zm0-2.5a.8.8 0 01.8.8h.2a1 1 0 00-1-1v.2zm-.8.8a.8.8 0 01.8-.8v-.2a1 1 0 00-1 1h.2zm.8.8a.8.8 0 01-.8-.8H9a1 1 0 001 1v-.2zm.8-.8a.8.8 0 01-.8.8v.2a1 1 0 001-1h-.2z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`Header renders with is divider 1`] = `
<div
  className="flex justify-around items-center w-full"
>
  <div
    className="sc-iXLiCy tpsuB"
  >
    <div
      className="sc-ixlHCg gxPqkx"
    >
      Title
    </div>
    <div
      className="sc-kEDQaP BGkxu"
    />
  </div>
</div>
`;
